docs/contributor/06-module-migration-guide.md

```diff
@@ -0,0 +1,231 @@
+# Module Migration Guideline
+
+This guide provides detailed instructions for how to migrate a Kyma module from existing (old) module metadata, especially channel-based ModuleTemplates, to the new module metadata, i.e. version-based ModuleTemplates accompanied by ModuleReleaseMeta.
+
+It is highly recommended that module teams familiarize themselves with the new module metadata before starting the migration:
+- The ADR backing the migration is [#984](https://github.com/kyma-project/community/issues/984)
+- An Update on Module Metadata has also been given in the 2024-11-26 Iteration Review
+
+## Target Process
+
+The target process is shown in the figure below. As an example, the `telemetry` module is used. The promotion of module versions and their assignments to channels is managed via three individual processes. In addition, there is a fourth process to delete a module version.
+
+![Target Process](./assets/module-migration.png)
+
+### 1) Submitting a new module version
+
+First, the module developer submits a new module version via a PR to the `/kyma/module-manifests` repository. The submission must provide a `module-config.yaml` file under the path `/modules/<module-name>/<module-version>` where `<module-version>` matches the version of the module configured in `module-config.yaml`. For the detailed format of the `module-config.yaml`, see below.
+
+Once the PR is opened, the submission pipeline verifies that the info is correct. E.g., it verifies that the provided `module-config.yaml` is valid (e.g., it verifies that the FQDN of the module doesn't change), it builds the module via `modulectl` in `--dry-run` mode, and it verifies that the version does not exist yet.
+
+Once the PR is merged, the submission pipeline builds and publishes the module via `modulectl` and pushes the generated ModuleTemplate to the `/kyma/kyma-modules` repository. The path of the generated ModuleTemplate is `/<module-name>/moduletemplate-<module-name>-<module-version>.yaml`.
+
+For more details, see [new submission pipeline](https://github.tools.sap/kyma/test-infra/blob/feature/new-submission-pipeline/ado/new-submission-pipeline-activity.md).
+
+> Note that this process only builds the necessary artifacts and puts them into their repositories, i.e. the OCI Registry and the `/kyma/kyma-modules` repository. The ModuleTemplate is NOT directly deployed into the KCP landscape.
+
+### 2) Submitting a channel mapping
+
+Second, the module developer submits a channel mapping via a PR to the `/kyma/module-manifests` repository. The submission must update the `module-releases.yaml` file under the path `/modules/<module-name>`. The `module-releases.yaml` is a simple mapping file to define what version each channel should map to. For the detailed format of the `module-releases.yaml`, see below.
```

Comment:
Does submitting the channel mapping mean submitting a `module-releases.yaml` file?
Where below? To the Migration Path section?